CliFx/Domain/ErrorTextWriter.cs

```diff
@@ -0,0 +1,138 @@
+using System;
+using System.Text.RegularExpressions;
+
+namespace CliFx.Domain
+{
+    internal class ErrorTextWriter
+    {
+        private const int indent = 4;
+
+        private static readonly ConsoleColor NameColor = ConsoleColor.DarkGray;
+        private static readonly ConsoleColor SpecificNameColor = ConsoleColor.White;
+        private static readonly ConsoleColor MessageColor = ConsoleColor.Red;
+        private static readonly ConsoleColor MethodColor = ConsoleColor.Yellow;
+        private static readonly ConsoleColor ParameterTypeColor = ConsoleColor.Blue;
+        private static readonly ConsoleColor FileColor = ConsoleColor.Yellow;
+        private static readonly ConsoleColor LineColor = ConsoleColor.Blue;
+
+        private static readonly Lazy<Regex> MethodMatcher = new Lazy<Regex>(() => new Regex(@"(?<prefix>\S+) (?<name>.*?)(?<methodName>[^\.]+)\("));
+        private static readonly Lazy<Regex> ParameterMatcher = new Lazy<Regex>(() => new Regex(@"(?<type>.+? )(?<name>.+?)(?:(?<separator>, )|\))"));
+        private static readonly Lazy<Regex> FileMatcher = new Lazy<Regex>(() => new Regex(@"(?<prefix>\S+?) (?<path>.*?)(?<file>[^\\/]+?(?:\.\w*)?):[^:]+? (?<line>\d+)"));
```

Comment:
Regex are already pretty much lazy unless you compile them, so there's no reason in doing this.